bluemira/equilibria/optimisation/problem/base.py

```diff
@@ -140,8 +140,8 @@ def read_coilset_state(
         Number of substates (blocks) in the state vector.
         """
         substates = 3
-        x, z = coilset.position
-        currents = coilset.current / current_scale
+        x, z = coilset.get_control_coils().position
```

Comment:
I'm slightly concerned that if this is being called when reading the coilset state, the set_coilset_state method below doesn't set the state on the control coils only, just sets the coilset currents to the state vector passed in.

But wouldn't that break? Wouldn't they be different array lengths?

Reply:
Is this related to loading an EQDSK with no coils? This changes this function to reading the coilset state, to reading the control coilset state. Separate function? Indexing after the fact?

Reply:
Sorry, this should not be on this branch. I am removing it.